src/mame/drivers/mc10.cpp

```diff
@@ -669,42 +504,67 @@ void mc10_state::mc10(machine_config &config)
 	m_maincpu->in_p2_cb().set(FUNC(mc10_state::mc10_port2_r));
 	m_maincpu->out_p2_cb().set(FUNC(mc10_state::mc10_port2_w));
 
-	/* video hardware */
-	SCREEN(config, "screen", SCREEN_TYPE_RASTER);
-
-	mc6847_ntsc_device &vdg(MC6847_NTSC(config, "mc6847", XTAL(3'579'545)));
-	vdg.set_screen("screen");
-	vdg.input_callback().set(FUNC(mc10_state::mc6847_videoram_r));
-
 	/* sound hardware */
 	SPEAKER(config, "speaker").front_center();
 	DAC_1BIT(config, m_dac, 0).add_route(ALL_OUTPUTS, "speaker", 0.0625);
 
 	CASSETTE(config, m_cassette);
-	m_cassette->set_formats(coco_cassette_formats);
+	m_cassette->set_formats(alice32_cassette_formats);
 	m_cassette->set_default_state(CASSETTE_STOPPED | CASSETTE_SPEAKER_ENABLED | CASSETTE_MOTOR_ENABLED);
 	m_cassette->add_route(ALL_OUTPUTS, "speaker", 0.05);
 	m_cassette->set_interface("mc10_cass");
 
 	/* printer */
-	rs232_port_device &rs232(RS232_PORT(config, "rs232", default_rs232_devices, "printer"));
-	rs232.set_option_device_input_defaults("printer", DEVICE_INPUT_DEFAULTS_NAME(printer));
+	rs232_port_device &rs232(RS232_PORT(config, "rs232", default_rs232_devices, "rs_printer"));
+	rs232.set_option_device_input_defaults("rs_printer", DEVICE_INPUT_DEFAULTS_NAME(printer));
+}
 
+void mc10_state::mc10_video(machine_config &config)
+{
 	/* internal ram */
-	RAM(config, m_ram).set_default_size("20K").set_extra_options("4K");
+	RAM(config, m_ram).set_default_size("4K").set_extra_options("8K,20K,32K");
+
+	/* video hardware */
+	SCREEN(config, "screen", SCREEN_TYPE_RASTER).set_raw(9.828_MHz_XTAL / 2, 320, 0, 320, 243, 0, 243);
+
+	mc6847_ntsc_device &vdg(MC6847_NTSC(config, "mc6847", XTAL(3'579'545)));
+	vdg.set_screen("screen");
+	vdg.input_callback().set(FUNC(mc10_state::mc6847_videoram_r));
+}
+void mc10_state::mc10(machine_config &config)
```

Comment:
Any specific reason for jamming these together?  We almost invariably have at least one blank line between functions.